app/scripts/lib/metaRPCClientFactory.js

```diff
@@ -0,0 +1,79 @@
+import { EventEmitter } from 'events';
+import { EthereumRpcError } from 'eth-rpc-errors';
+import createRandomId from '../../../shared/modules/random-id';
+
+const metaRPCClientFactory = (connectionStream) => {
```

Comment:
this is the same as the `MetaRPCClient`, I just have both of them up to show side-by-side what it would look like as a Factory vs class as per @Gudahtt 's feedback

Reply:
We can choose which one we like the look of better, I also have the tests up for both versions

Reply:
Oops - when I suggested a factory method, I meant one that constructed the class instance for you! e.g.

```
createMetaRpcClient() {
  const metaRPCClient = new MetaRPCClient(streamTest);
  return new Proxy(metaRPCClient, {
    get: (object, property) => {
      if (object[property]) {
        return object[property];
      }
      return (...p) => {
        const cb = p[p.length - 1];
        const params = p.slice(0, -1);
        const id = createRandomId();
    
        object.requests.set(id, cb);
        object.connectionStream.write({
          method: property,
          params,
          id,
        });
      };
    },
  });
}
```

Then we could make that the only module export to ensure the constructor wasn't used directly.

This version seems a bit more difficult to read :thinking: For me at least. Using a class seemed appropriate.